### ISSUE TITLE
Support "ls" command and endpoint definitions

### ISSUE THREAD

**Issue by michael-lazar** [eliangcs/http-prompt#63]
I've been using this program for a few days now and overall I love it. I'm using it to test endpoints while developing my API.

One issue I have however is that whenever I type `cd`, my muscle memory kicks in and I immediately follow with an `ls` out of habit. It's always a little disappointing when this doesn't work :( I was thinking of ways to solve this, and it would be great if I could pass in my endpoint definition file ala

``` bash
$ http-prompt --specification=my_api_spec.json
```

This would enable http-prompt to do a bunch of smart things like suggest endpoints when I type `ls` or  autocomplete parameter names. Could also add a `help` command to view the documentation for any given endpoint. There are a couple of widely used spec formats, with Swagger probably being the most popular

http://nordicapis.com/top-specification-formats-for-rest-apis/

**Comment by eliangcs** [comment]
I like the `ls` idea. I myself enter `ls` by mistake a lot, too. `help` command is already there for listing all the available commands, so I think it's better to keep it that way. Maybe we could use `man` or something like that to print the documentation of an endpoint from a spec.

My original idea for `ls` is to list body parameters, querystring parameters, and headers. Are you suggesting the same?

**Comment by michael-lazar** [comment]
I was actually thinking more along the lines of treating the api as a file structure and using ``ls`` for navigation. So if you define your endpoints like this

http://website.com/
http://website.com/users/admin
http://website.com/users/{uid}
http://website.com/about
http://website.com/status

```bash
http://website.com> ls
about    users   status
http://website.com> cd users
http://website.com/users> ls
{uid}   admin
```

**Comment by eliangcs** [comment]
This feature has been added in v0.10.0, which is only available in paid subscription for now. See http://http-prompt.com/early-access for more info.

**Comment by eliangcs** [comment]
I decided to remove the paid subscription, so this feature is publicly available now in v0.10.0.
